numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4088,6 +4088,7 @@ initialize_numeric_types(void)
      * #NAME = Byte, Short, Int, Long, LongLong, UByte, UShort,
      *         UInt, ULong, ULongLong#
      */
+    @name@_arrtype_as_number = gentype_as_number;
```

Comment:
On my phone so haven't looked at the context here, but do I infer correctly from the commit message that this is just filling in some arbitrary pointer here that will be overwritten later? If so then there should be a comment to that effect here, because otherwise to a naive reader it looks like there's some actual reason why gentype_as_number will work.